### PR TITLE
build(snap): update snap to core24

### LINKED ISSUE #750: Update the rockcraft snap to core24
### What needs to get done

Update the rockcraft snap to core24

### Why it needs to get done

To keep up with the times

### PR DESCRIPTION
**Pull request by tigarmo**
Fixes #750

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -65,15 +74,14 @@ parts:
...
-      - "usr/lib/x86_64-linux-gnu/liblber*"
-      - "usr/lib/x86_64-linux-gnu/libsasl2*"
+      - "-usr/bin"
+      - "-usr/share/doc"
+      - "-usr/share/man"
+      # perl is part of the core22 / core24
+      - "-usr/share/perl"
+      - "-usr/lib/x86_64-linux-gnu/perl"
+      - "-usr/lib/x86_64-linux-gnu/libperl*"
+      - "-usr/lib/x86_64-linux-gnu/libgdbm*"
```
Comment (dariuszd21) [inline snap/snapcraft.yaml:84]: Previous version has some missing library for `core24` ?

**Review by dariuszd21 — APPROVED** [review]
Thank you!

Is it something that should be included in changelog ?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (tigarmo) [reply snap/snapcraft.yaml:84]: yes, it was complaining about `libssh.so.0.4`

**Comment by tigarmo** [comment]
> Thank you!
> 
> Is it something that should be included in changelog ?

We're not currently updating the changelog with individual PRs in rockcraft. I want to start doing that next year (maybe) but for now the changelog is a separate task pre-release.

**Review by mr-cal — COMMENTED** [review]
Do you need to update the github workflows to test on 3.12?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -65,15 +74,14 @@ parts:
...
-      - "usr/lib/x86_64-linux-gnu/libpsl*"
-      - "usr/lib/x86_64-linux-gnu/librtmp*"
-      - "usr/lib/x86_64-linux-gnu/libnghttp2*"
-      - "usr/lib/x86_64-linux-gnu/libldap*"
-      - "usr/lib/x86_64-linux-gnu/liblber*"
-      - "usr/lib/x86_64-linux-gnu/libsasl2*"
+      - "-usr/bin"
+      - "-usr/share/doc"
+      - "-usr/share/man"
+      # perl is part of the core22 / core24
```
Comment (sergiusens) [inline snap/snapcraft.yaml:80]: are we using perl itself from the core24 base? If so, that could be a problem as it is not `classic` ready

**Review by sergiusens — APPROVED** [review]

File: tests/spread/rockcraft/plugin-python/src/setup.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from distutils.core import setup
+from setuptools import setup
```
Comment (sergiusens) [inline tests/spread/rockcraft/plugin-python/src/setup.py:1]: this worries me a bit because it is a test

**Pull request merged**